### ISSUE TITLE
twice action when using @ operator at version 0.3.0

### ISSUE THREAD

**Issue by sv-atoslav** [coriolinus/calc#14]
### my program and OS versions
```
$calc --version
calc 0.3.0
$lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```
### steps to reproduce 1
```
$calc --i64
[1]: 528500/100
5285
[2]: @/2
1321
[3]: 5285/2
2642
[4]:
Error: Interrupted
```
### steps to reproduce  2
```
$calc --u64
[0]: 2+2*2
6
[1]: @*100
60000
[2]: @*-1
operation `-` not implemented for u64
[2]: @*2
2400
[3]: @+1
1202
[4]:
Error: Interrupted
```
### steps to reproduce 3
```
[0]: 0.1+0.2
.30000000000000004 (skip, because https://0.30000000000000004.com/ )
[1]: @+0.003
.30600000000000005
[2]: @+0.0001
.3032
[3]: 
Error: Interrupted
```

**Comment by coriolinus** [comment]
Interesting! Thanks for reporting. I've added some tests in https://github.com/coriolinus/calc/commit/cf8f2305f660d5cad9954e4f99b7f3cc4993e061, which demonstrate that the error is not in the core context/evaluation engine. However, I've successfully repro'd the issue locally. Therefore it looks like the operation is getting evaluated twice somehow. I will continue to investigate.

**Comment by coriolinus** [comment]
Fixed in #15. Thanks for reporting this issue!
